python/ql/src/Expressions/WrongNumberArgumentsForFormat.ql

```diff
@@ -15,29 +15,29 @@
 import python
 import semmle.python.strings
 
-predicate string_format(BinaryExpr operation, StrConst str, Object args, AstNode origin) {
-    exists(Object fmt, Context ctx | operation.getOp() instanceof Mod |
-           operation.getLeft().refersTo(ctx, fmt, _, str) and
-           operation.getRight().refersTo(ctx, args, _, origin)
+predicate string_format(BinaryExpr operation, StrConst str, Value args, AstNode origin) {
+    exists(Value fmt, Context ctx | operation.getOp() instanceof Mod |
```

Comment:
It's a bit odd that the check on `operation` is done within the `exists`. Might as well move it outside while you're in the neighbourhood.

Reply:
Done!